### PR TITLE
Ports filter surgery

### PR DESCRIPTION
**Pull request by maxion12345**
Port: https://github.com/BeeStation/BeeStation-Hornet/pull/2603

# General Documentation

### Intent of your Pull Request

Adds two things:
The Blood Filter surgical tool that basically exists to filter blood
and the Filter Blood surgery that exists to remove chemicals from people actively, rather then relying on passive chemical removal. Only appears when there are reagents in the body. 

### Why is this change good for the game?
you ever get those guys dosed with absurd amounts of chems and go through the cycle of "revive, let him process a bit of chem out of his system, he dies from the chem, repeat"? thats why this change is good.
# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 
wiki documentation is that the surgery is scalpel, retractor, scalpel, then blood filter (auto-repeats) , then cautery

### What should players be aware of when it comes to the changes your PR is implementing?
the players can look up a bit and read the shit I already wrote
### What general grouping does this PR fall under? 
look at the changelog

### If there are any numerical values involved in your PR that will be relevant to a player, please note them here. 
The more chem you have in you, the faster it gets removed. There are diminishing returns, so you'll want to pair this with traditional methods to achieve the fastest chem removal

# Changelog

Edit this changelog for changes that are noticeable by the players. Remove it if this isn't the case. If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name. Prefix the PR title with [admin] if it's something admin related. Prefix the PR title with [s] if you are fixing an exploit so it is not announced on discord and the server.

:cl:  Chessus
rscadd: Added the blood filter item and surgery for when you need to revive the assistant with 300 of each alcoholic drink in him, or for when dipshit mcdoctorperson spams every conceivable chem on the sleeper
/:cl:

### CODE REVIEW

File: code/modules/surgery/blood_filter.dm [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	if(HAS_TRAIT(target, TRAIT_HUSK) || target.reagents.total_volume==0) //Can't filter husk or 0 regent body
+		return FALSE
+	return ..()
+
+/datum/surgery_step/filter_blood/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	if(istype(surgery,/datum/surgery/blood_filter))
+		var/datum/surgery/blood_filter/the_surgery = surgery
+		if(!the_surgery.antispam)
+			display_results(user, target, "<span class='notice'>You begin filtering [target]'s blood...</span>",
+		"<span class='notice'>[user] uses [tool] to filtering your blood.</span>",
```
Comment (SomeguyManperson) [inline code/modules/surgery/blood_filter.dm]: this line makes me want to die

**Comment by SomeguyManperson** [comment]
reference the original pr in the description somewhere i.e. these two bits
image
then # and the pr's number except the organization and repository from where the pr was ported from

File: code/modules/surgery/blood_filter.dm (hunk 1 above)
Comment (maxion12345) [reply code/modules/surgery/blood_filter.dm]: i don't know enough about coding to know why

Comment (adamsong) [reply code/modules/surgery/blood_filter.dm]: “Uses blood filter to filtering your blood”

Comment (maxion12345) [reply code/modules/surgery/blood_filter.dm]: ahf uck i see it now

**Review by nesisknesis — APPROVED** [review]
awful

also if its a port you generally put the original author's name here so that they get credited in the changelog 
image

**Comment by MegaEmpirical** [comment]
How does this work with plasmamen?

**Comment by maxion12345** [comment]
> How does this work with plasmamen?

Just like other surgeries, you'll want them under a shower before you begin. If the amount of chems is large enough to warrant doing this surgery then it shouldn't take too many steps to bring them down to a level you can manage with just chems.

**Comment by adamsong** [comment]
Plasmemes don't have blood though

**Comment by maxion12345** [comment]
It works the same way, I suppose we can change it so it filters the "plasma"? Not even sure how plasma people retain chemicals if they are skeletons. Bone filtering surgery, I guess.

guess thats a to-do, to make it so it either doesn't work on plasmapeople or so that it says "filter plasma" instead of blood.

**Comment by maxion12345** [comment]
Closing because I've lost the will to do further work on a very situational surgery

**Comment by N3D6** [comment]
F

**Pull request closed without merging**